src/value/owned.rs

```diff
@@ -9,19 +9,20 @@ use super::{AnyInt, Element, ImportSource, Sequence, Struct, SymbolToken};
 use crate::types::SymbolId;
 use crate::IonType;
 use std::collections::HashMap;
+use std::rc::Rc;
 
 /// An owned implementation of  [`ImportSource`].
 #[derive(Debug, Clone)]
 pub struct OwnedImportSource {
     // TODO consider Rc<String> here as we flesh out sharing/symbol tables.
-    table: String,
+    table: Rc<String>,
     sid: SymbolId,
 }
 
 impl OwnedImportSource {
     pub fn new<T: Into<String>>(table: T, sid: SymbolId) -> Self {
         Self {
-            table: table.into(),
+            table: Rc::new(table.into()),
```

Comment:
I think we need to change the generic here to take `Into<Rc<String>>` or something otherwise it will be impossible to share from an name that is itself an `Rc<String>`.  That said `Into<Rc<String>>` may not be as ergonomic with respect to `&str`, so your mileage may vary...